lib/screens/app/wallet/scan.dart

```diff
@@ -1,210 +1,132 @@
 
-  Future scan() async {
-    setState(() {
-      this.step = Steps.scan;
-    });
+  void processAction(dynamic action) async {
+    final data =
+        await EosService.of(context, listen: false).fillActionPlaceholders(
+      account: action["account"],
+      name: action["name"],
+      data: action["data"],
+    );
 
-    try {
-      ScanResult scanResult = await BarcodeScanner.scan();
-      if (scanResult.type == ResultType.Cancelled) {
-         Navigator.of(context).pop();
-         
-      } else {
-        setState(() {
-          this.step = Steps.processing;
-          this.qrcode = scanResult.rawContent;
-        });
-        processSigningRequest();
-      }
-    } on PlatformException catch (e) {
-      if (e.code == BarcodeScanner.cameraAccessDenied) {
-        setState(() {
-          this.error = 'The user did not grant the camera permission!';
-          this.step = Steps.error;
-        });
-      } else {
-        setState(() {
-          this.error = 'Unknown error: $e';
-          this.step = Steps.error;
-        });
-      }
-    } on FormatException {
-      setState(() {
-        this.error =
-            'null (User returned using the "back"-button before scanning anything. Result)';
-        this.step = Steps.error;
-      });
-    } catch (e) {
-      setState(() {
-        this.error = 'Scan unknown error: $e';
-        this.step = Steps.error;
-      });
-    }
+    NavigationService.of(context).navigateTo(
+      Routes.customTransaction,
+      CustomTransactionArguments(
+        account: action["account"],
+        name: action["name"],
+        data: data,
+      ),
+    );
   }
 
-  void processSigningRequest() async {
-    try {
+  void qrCodeCallback(String code) async {
+    if (this.step == Steps.processing || this.step == Steps.success) return;
 
-      print("QR Code: "+this.qrcode);
+    setState(() {
+      this.step = Steps.processing;
+    });
 
-      final esr = await EosioSigningRequest.factory(
-        EosService.of(context, listen: false).client,
-        this.qrcode,
-        SettingsNotifier.of(context, listen: false).accountName,
-      );
+    try {
+      final esr = SigningRequestManager.from(code,
+          options: defaultSigningRequestEncodingOptions(
+            nodeUrl: EosService.of(context, listen: false).baseURL,
+          ));
 
-      assert(esr.action.account.isNotEmpty);
-      assert(esr.action.name.isNotEmpty);
+      final type = esr.data.req[0];
 
       setState(() {
         this.step = Steps.success;
       });
 
-      Map<String, dynamic> data = Map<String, dynamic>.from(esr.action.data);
-
-      NavigationService.of(context).navigateTo(
-          Routes.customTransaction,
-          CustomTransactionArguments(
-            account: esr.action.account,
-            name: esr.action.name,
-            data: data,
-          ),
-          true);
+      switch (type) {
+        case 'identity':
+          processIdentity(esr.data);
```

Comment:
Does this mean it's now successfully handling identity? Just a question. This is not a regression as it didn't do that before.

Reply:
processIdentity is empty function currently, but it has to be implemented in the way to redirect user to another screen and execute callback after confirmation